### PR TITLE
Initial context setup

### LINKED ISSUE #14: Initialize Context structure
Add context to support `useContext` hooks for semi-global state
QuestionContext and AnswerContext
QuestionContext - Initialized through API get on app load, contains all questions and relevant data
AnswerContext - The conglomeration of all answers, will be used to send post request on `generateForm` button

### LINKED ISSUE #25: Implement functionality for question and answer context
* Question context will be relevant from app -> survey
* Answer context will be relevant from survey -> questions

### PR DESCRIPTION
**Pull request by BlakeDonn**
### What does this PR do?

* Adds capability to manage state in two different contexts
* Adds QuestionContext
* Adds AnswerContext
* Allows for state to be passed without utilizing props
* Introduces types and interfaces for dummy api calls

### Where should the reviewer start?
* Check that types are correctly implemented, check that context cover the required scope

### How should this be manually tested?
* In order to test the api call you will need an outside resource that hosts .json files for mock api functionality 
* [link](https://my-json-server.typicode.com/typicode/demo)

### Any background context you want to provide?
* Once we get the real endpoint running, it should be simple to plug and play with this code
* Currently researching how to update context, or if there should be another way to go about it.

### What are the relevant tickets?

closes #25 
closes #14

### Screenshots (if appropriate)

QuestionContext: 
Screen Shot 2021-01-06 at 7 25 04 PM

AnswerContext: 
Screen Shot 2021-01-06 at 7 57 53 PM

### CODE REVIEW

File: src/components/Home/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+export const Home: React.FC = () => {
+  return (
+    <div>
+      {"hi"}
```
Comment (Arique1104) [inline src/components/Home/Home.tsx]: hello!

File: src/components/Question/Question.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React  from 'react';
+
+export const Question: React.FC = () => {
+  return (
+    <div>
+      {"hi"}
```
Comment (Arique1104) [inline src/components/Question/Question.tsx:5]: hello!

File: .eslintcache [hunk 3]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+[{"/Users/bdizzle/Desktop/codebases/4-mod/fee-dreamhome/src/components/Sourvey/Sourvey.tsx":"1"},{"size":934,"mtime":1609981698141,"results":"2","hashOfConfig":"3"},{"filePath":"4","messages":"5","errorCount":1,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"source":null},"13szocz","/Users/bdizzle/Desktop/codebases/4-mod/fee-dreamhome/src/components/Sourvey/Sourvey.tsx",["6"],{"ruleId":null,"fatal":true,"severity":2,"message":"7","line":1,"column":0},"Parsing error: Merge conflict marker encountered."]
```
Comment (Arique1104) [inline .eslintcache]: Does this change every time you have a new person driving?

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -1,44 +1,30 @@
...
-// import Home from '../Home/Home';
-// import Error from '../Error/Error';
-// import Journey from '../Journey/Journey';
-import Sourvey from '../Sourvey/Sourvey';
-import NavBar from '../NavBar/NavBar';
-// import GenerateReport from '../GenerateReport/GenerateReport';
+import Home from '../Home/Home';
+//import Error from '../Error/Error';
+//import Journey from '../Journey/Journey';
+import {Sourvey} from '../Sourvey/Sourvey';
```
Comment (Arique1104) [inline src/components/App/App.tsx:9]: We should update this to 'Survey'

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -1,44 +1,30 @@
...
-
-  const getQuestionsData = () => {
-
-    // setData();
-  }
+const App:React.FC = () =>{
+  const [questions, updateQuestions] = useState<AllQuestionFormat | null>(null);
+  useEffect(() => {
+    getQuestions().then((data) => updateQuestions(data))
+  }, []);
```
Comment (Arique1104) [inline src/components/App/App.tsx:19]: Are you loading up the questions dynamically so that they have the same effect?

File: .eslintcache (hunk 3 above)
Comment (Atos20) [inline .eslintcache]: I don't like how this file is changing every time we switch from one branch to another.

File: src/components/App/App.css [hunk 6]
```diff
@@ -1,5 +1,38 @@
-.name{
-  font-size: 2em;
-  height: 5em;
-  background-color: red;
-}
\ No newline at end of file
+.App {
```
Comment (Atos20) [inline src/components/App/App.css:1]: I had removed this already and your PR is bringing it back. Not a big deal but something to keep an eye on.  There is something definitely with rebase that I don't get.

**Review by Arique1104 — APPROVED** [review]
This is awesome!

File: src/components/App/App.tsx [hunk 7]
```diff
@@ -1,44 +1,30 @@
...
-    */}
-    </>
-    );
-}
+    <QuestionContext.Provider value={questions}>
+      <div>
+        <Home />
+        <Sourvey />
+      </div>
+     </QuestionContext.Provider>
```
Comment (Atos20) [inline src/components/App/App.tsx:27]: Just to make sure, as long as the component is wrapped within the provider  "tag" they will have access to the state, correct?

File: src/components/Sourvey/Sourvey.tsx [hunk 8]
```diff
@@ -1,9 +1,28 @@
-import React from 'react'
+import React, {useEffect, useContext, useState} from 'react';
+import {AllQuestionFormat, AnswerContext, QuestionContext} from '../../types';
```
Comment (Atos20) [inline src/components/Sourvey/Sourvey.tsx:2]: we just need to import it :octocat:

**Review by Arique1104 — APPROVED** [review]
Ignore this. I pressed a button by accident.

**Review by Atos20 — APPROVED** [review]

File: .eslintcache (hunk 3 above)
Comment (timkeresey) [reply .eslintcache]: I'm not sure what I did, but it's not an issue for me anymore and not injecting itself into every commit. We can try to figure it out when we meet.

File: src/components/App/App.tsx (hunk 5 above)
Comment (BlakeDonn) [reply src/components/App/App.tsx:19]: useEffect is a bit misleading, this is essentially just a fetch for the data ( currently mocked on a temporary endpoint) that happens when the application is fired up. 

But yeah! This fetch will get all the questions and store them locally so we can display them to the DOM

File: src/components/Question/Question.tsx (hunk 2 above)
Comment (Atos20) [reply src/components/Question/Question.tsx:5]: oh! hello

File: .eslintcache (hunk 3 above)
Comment (Atos20) [reply .eslintcache]: we are not doing driver/navigator but this changes every time we checkout to a new or existing branch 😭

File: src/components/App/App.tsx (hunk 7 above)
Comment (BlakeDonn) [reply src/components/App/App.tsx:27]: Yep! And the `value={}` of the provider is essentially where the information lives. In this case the value is state, so if that state is updated on App, all children will have access to the updated info and re-render if the diffing algo determines it neccessary.

File: src/components/App/App.tsx (hunk 4 above)
Comment (Atos20) [reply src/components/App/App.tsx:9]: Thanks for the catch!

File: src/components/App/App.tsx (hunk 5 above)
Comment (timkeresey) [reply src/components/App/App.tsx:19]: We'll be loading them up here through the API call, then displaying them with a map in another component.

File: src/components/App/App.tsx (hunk 7 above)
Comment (timkeresey) [reply src/components/App/App.tsx:27]: Only App will be wrapped in the provider tag, which gives every component rendered within that wrapper access to the context state via accessing useContext.

File: src/components/Question/Question.tsx (hunk 2 above)
Comment (timkeresey) [reply src/components/Question/Question.tsx:5]: goodbye

File: src/components/App/App.tsx (hunk 5 above)
Comment (Atos20) [reply src/components/App/App.tsx:19]: `useEffect` is just a built-in function in React that fires `onload`. This way will have the data available to us ready to display to the user.

 If your question is if `useEffect` adds a visual effect to each question, then I am afraid to tell that it is just the basic functionality to display data.

**Review by timkeresey — APPROVED** [review]

**Pull request merged**